### PR TITLE
#126 lighthouse on ci

### PR DESCRIPTION
**Pull request by adamdavies1**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Please also include any acceptance criteria if you have any.

- Lighthouse for runner and designer
- Changes to components.json to match current schema layouts (removing list out of options)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- [ ] You can run the Branch QA workflow on this branch to see the Run Lighthouse CI section working. Or you can run the command `npm install -g @lhci/cli@0.8.x` followed by `lhci autorun` locally
# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and versioning
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto main and there are no code conflicts
- [x] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Comment by adamdavies1** [comment]
pipeline is failing for this branch due to the issues discovered by running lighthouse. Perhaps we need to raise a new task to fix these?

**Comment by adamdavies1** [comment]
@jenbutongit lots of errors on lighthouse for this one but I think they are performance errors. Are we only using lighthouse for accessibility?

**Comment by jenbutongit** [comment]
@adamdavies1 - I'm pretty sure for the runner I had lighthouse only running accessibility metrics. We could do more - but I'm aware designer probably would need a lot of work in that region

**Comment by adamdavies1** [comment]
> @adamdavies1 - I'm pretty sure for the runner I had lighthouse only running accessibility metrics. We could do more - but I'm aware designer probably would need a lot of work in that region

@jenbutongit nice green pipeline now, ready for review

File: lighthouserc.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      startServerCommand: "",
+      settings: {
+        //set which categories you want to run here.
+        onlyCategories: ["accessibility"],
+      },
+    },
+    assert: {},
+    upload: {
+      target: "filesystem",
+      //target: "temporary-public-storage",
```
Comment (JackReevies) [inline lighthouserc.js:20]: Should this be removed?

File: runner/src/server/forms/components.json [hunk 2]
```diff
@@ -11,109 +11,91 @@
           "name": "textField",
           "title": "Text field",
           "hint": "Help text",
-          "options": {},
```
Comment (endamccormack) [inline runner/src/server/forms/components.json:14]: why did you take out these options?

File: lighthouserc.js (hunk 1 above)
Comment (endamccormack) [reply lighthouserc.js:20]: should be removed yeah

**Review by endamccormack — CHANGES_REQUESTED** [review]

File: lighthouserc.js (hunk 1 above)
Comment (adamdavies1) [reply lighthouserc.js]: removed this

File: runner/src/server/forms/components.json (hunk 2 above)
Comment (jenbutongit) [reply runner/src/server/forms/components.json:14]: please don't remove options from the schema. the required flags live there! But the corrections to the list definitions are correct.

any changes to schema I think I'd like to see an ADR or explanation in your PR please

**Review by jenbutongit — CHANGES_REQUESTED** [review]
please keep the options.required flag where it is. only `list: ..` has been hoisted out of options

File: runner/src/server/forms/components.json (hunk 2 above)
Comment (adamdavies1) [reply runner/src/server/forms/components.json]: sorry @jenbutongit, must of gotten my wires crossed. I have added back the options and kept the correction to lists .

**Review by jenbutongit — APPROVED** [review]
LGTM! thanks! 🎉

**Pull request merged**